Yoast/wordpress-seo
# Move permalink notification to Site Health.

## Summary


This PR can be summarized in the following changelog entry:

* Moved the postname in permalink notification from the SEO Dashboard to WordPress' Health Check

## Relevant technical choices:

- Added two deprecations and one upgrade routine using a temporary xx.x for the plugin version. This should be updated when merging into trunk once the actual milestone will be determined.
- Better wording for the non-passed / passed test welcome.


## Test instructions

This PR can be tested by following these steps:

- Go to Settings > Permalink and set your permalink structure to "Plain" or "Numeric" or a custom one that does not contain `/%postname%/` 
- Save
- Go to the Site Health page: Tools > Site Health and see a check being presented:

<img width="841" alt="Screenshot 2019-12-30 at 12 36 01" src="https://user-images.githubusercontent.com/1682452/71580470-1d9cd300-2b01-11ea-80ec-6074ee78f645.png">

- Go to Settings > Permalink and set your permalink structure to "Post name" or "Day and name" or "Month and name" or a custom one that does contain `/%postname%/` 
- Save
- Go to the Site Health page and see the check is now within the "Passed tests" section:

<img width="838" alt="Screenshot 2019-12-30 at 12 36 34" src="https://user-images.githubusercontent.com/1682452/71580476-27263b00-2b01-11ea-8c4a-91dcb25b7577.png">


## UI changes

## Documentation

## Quality assurance


Fixes https://github.com/Yoast/wordpress-seo/issues/12900
